### ISSUE TITLE
Again occasional crashes on init due to corrupt config

### ISSUE THREAD

**Issue by mpters** [Tealium/tealium-kotlin#203]
Hi,

it seems like issue #175 or a similar problem occurs again with tealium-core version 1.5.1. Even earlier versions might be affected since I can see this crash occasionally occur within the last three month.

I wonder if we should wrap the initialization in a try/catch block? The crash prevents some users from using our app at all because it crashes at start up. And obviously the fix of #175 was only temporary and I wonder how we can solve this problem once for all. I'm looking forward to hear your suggestions.

FYI: We're talking about the same project as in #175. The author of the original issue is a colleague of mine.

**Stacktrace:**
```
Caused by org.json.JSONException: Value  of type java.lang.String cannot be converted to JSONObject
       at org.json.JSON.typeMismatch(JSON.java:112)
       at org.json.JSONObject.<init>(JSONObject.java:172)
       at org.json.JSONObject.<init>(JSONObject.java:185)
       at com.tealium.core.settings.b.a(SourceFile:15)
       at com.tealium.core.settings.b.d(SourceFile:4)
       at com.tealium.core.settings.b.<init>(SourceFile:24)
       at com.tealium.core.settings.b.<init>(SourceFile:159)
       at com.tealium.core.Tealium.<init>(SourceFile:30)
       at com.tealium.core.Tealium.<init>(SourceFile)
       at com.tealium.core.Tealium$Companion.create(SourceFile:1)
       at xxxxx.data.external.analytics.TealiumWrapper.init(TealiumWrapper.kt:107)
       at xxxxx.data.external.analytics.AnalyticsWrapper.<init>(AnalyticsWrapper.kt:20)
       at xxxxx.di.module.ApplicationModule.provideAnalyticsWrapper$XXXXX_4_8_0_release(ApplicationModule.kt:507)
       at xxxxx.di.module.ApplicationModule_ProvideAnalyticsWrapper$XXXXX_4_8_0_releaseFactory.provideAnalyticsWrapper$XXXXX_4_8_0_release(ApplicationModule_ProvideAnalyticsWrapper$XXXXX_4_8_0_releaseFactory.java:56)
       at xxxxx.DaggerXXXXXApplication_HiltComponents_SingletonC$SingletonCImpl$SwitchingProvider.get0(DaggerXXXXXApplication_HiltComponents_SingletonC.java:5943)
       at xxxxx.DaggerXXXXXApplication_HiltComponents_SingletonC$SingletonCImpl$SwitchingProvider.get(DaggerXXXXXApplication_HiltComponents_SingletonC.java:6325)
       at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
       at xxxxx.DaggerXXXXXApplication_HiltComponents_SingletonC$SingletonCImpl.injectXXXXXApplication2(DaggerXXXXXApplication_HiltComponents_SingletonC.java:5640)
       at xxxxx.DaggerXXXXXApplication_HiltComponents_SingletonC$SingletonCImpl.injectXXXXXApplication(DaggerXXXXXApplication_HiltComponents_SingletonC.java:5465)
       at xxxxx.Hilt_XXXXXApplication.hiltInternalInject(Hilt_XXXXXApplication.java:50)
       at xxxxx.Hilt_XXXXXApplication.onCreate(Hilt_XXXXXApplication.java:41)
       at xxxxx.XXXXXApplication.onCreate(XXXXXApplication.kt:83)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1266)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7632)
       at android.app.ActivityThread.-$$Nest$mhandleBindApplication(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2402)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8775)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

**TealiumConfig:**
```
        val tealiumCfg = TealiumConfig(
            appContext as Application,
            tealiumAccountName,
            tealiumProfileName,
            if (BuildConfig.DEBUG) Environment.QA else Environment.PROD,
            modules = mutableSetOf(Lifecycle, Modules.InstallReferrer),
            dispatchers = mutableSetOf(
                TagManagementDispatcher,
                Dispatchers.RemoteCommands
            )
        ).apply {
            overrideTagManagementUrl = tealiumUtagUrl
            overrideLibrarySettingsUrl = tealiumUtagUrl
            useRemoteLibrarySettings = true
            consentManagerPolicy = ConsentPolicy.GDPR
            consentManagerLoggingEnabled = true
            consentExpiry = ConsentExpiry(CONSENT_EXPIRY_DAYS, TimeUnit.DAYS)
            sessionCountingEnabled = false
        }
```

**Versions we're using:**
```
tealium-core = "1.5.1"
tealium-installreferrer = "1.1.0"
tealium-lifecycle = "1.1.1"
tealium-remotecommand = "1.2.1"
tealium-tagmanagement = "1.2.0"
```

Kind regards
Marcel

**Comment by jameskeith** [comment]
Hi @mpters 
thanks for flagging this with us again - sorry to hear that the previous update has not fully resolved the issue - obviously the checks in place were not quite sufficient, so we'll got some better exception handling in place for this code. 
I'll get a ticket raised internally to have this addressed asap.

**Comment by jameskeith** [comment]
@mpters - apologies for not notifying you of the fix for this, but the fix for this was released in `1.5.2` 
please let us know if you have any further issues.
